src/pages/profiles/ProfileList.tsx

```diff
@@ -1,46 +1,119 @@
 import { fetchProfiles } from "api/profiles";
-import BaseLayout from "components/BaseLayout";
+import classnames from "classnames";
 import { useQuery } from "@tanstack/react-query";
 import { queryKeys } from "util/queryKeys";
 import { useNotify } from "context/notify";
 import Loader from "components/Loader";
 import { Link, useNavigate, useParams } from "react-router-dom";
 import ItemName from "components/ItemName";
 import { getProfileInstances } from "util/usedBy";
+import usePanelParams from "util/usePanelParams";
+import { usePagination } from "util/pagination";
+import useEventListener from "@use-it/event-listener";
+import { updateTBodyHeight } from "util/updateTBodyHeight";
+import Pagination from "components/Pagination";
+import NotificationRow from "components/NotificationRow";
+import { fetchProject } from "api/projects";
 
 const ProfileList: FC = () => {
   const navigate = useNavigate();
   const notify = useNotify();
+  const panelParams = usePanelParams();
   const { project } = useParams<{ project: string }>();
+  const [query, setQuery] = useState<string>("");
 
   if (!project) {
     return <>Missing project</>;
   }
+  const isDefaultProject = project === "default";
 
   const {
     data: profiles = [],
     error,
-    isLoading,
+    isLoading: isProfilesLoading,
   } = useQuery({
     queryKey: [queryKeys.profiles, project],
     queryFn: () => fetchProfiles(project),
   });
 
+  const {
+    data: projectObj,
+    error: projectError,
+    isLoading: isProjectLoading,
+  } = useQuery({
+    queryKey: [queryKeys.projects, project],
+    queryFn: () => fetchProject(project),
+  });
+
   if (error) {
     notify.failure("Loading profiles failed", error);
   }
 
+  if (projectError) {
+    notify.failure("Loading project failed", error);
+  }
+  const isLoading = isProfilesLoading || isProjectLoading;
+
+  const featuresProfiles = projectObj?.config["features.profiles"] === "true";
+
+  const instanceCountMap = profiles.map((profile) => {
+    const usedByInstances = getProfileInstances(
+      project,
+      isDefaultProject,
+      profile.used_by
+    );
+    return {
+      name: profile.name,
+      count: usedByInstances.filter((instance) => instance.project === project)
+        .length,
+      total: usedByInstances.length,
+    };
+  });
+
+  const filteredProfiles = profiles.filter((item) => {
+    if (query) {
+      const q = query.toLowerCase();
+      if (
+        !item.name.toLowerCase().includes(q) &&
+        !item.description.toLowerCase().includes(q)
+      ) {
+        return false;
+      }
+    }
+    return true;
+  });
+
   const headers = [
     { content: "Name", sortKey: "name" },
     { content: "Description", sortKey: "description" },
-    { content: "Used by", sortKey: "used_by", className: "u-align--right" },
+    {
+      content: isDefaultProject ? (
+        <>
+          Total instances
+          <br />
+          <div className="header-second-row">Used by</div>
```

Comment:
The split heading is confusing, can we change it to just say "Used by" and clarify the two numbers in the cells?

Reply:
This is a question for @al-bakalova

Reply:
> The split heading is confusing, can we change it to just say "Used by" and clarify the two numbers in the cells?

Once we switch these, it is getting confused, I agree. What if we keep only "Used by" and the have:
10 instances
100 in all projects 

But then we sort by the number of instances in the current project?

Reply:
Sure, it works for me.